File: py/soal4.py
# ANGGOTA KELOMPOK
# - Naufal Fahrezi (14081020035)
# - Amalia Nur Fitri (140810200045)
# - Calvin Calfi Montolalu (140810200053)
# - Deani Puteri Virdiana (140810200059)

class PersegiPanjang :
    def __init__(self):
        self.__panjang=0
        self.__lebar=0

    def setPersegiPanjang(self, p, l) :
        self.__panjang=p
        self.__lebar=l

    def setPanjang(self, p) : self.__panjang=p
    def setLebar(self, l) : self.__lebar=l

    def getPanjang(self) : return self.__panjang
    def getLebar(self) : return self.__lebar

    def hitungLuas(self) : return int(self.__panjang) * int(self.__lebar)

    def input(self) :
        self.__panjang=input("Masukkan panjang: ")
        self.__lebar=input("Masukkan lebar: ")

    def print(self) :
        print("Panjang = " + str(self.__panjang))
        print("Lebar = " + str(self.__lebar))

    def printLuas1(self, luas) :
        print("Luas = " + str(luas))
    
    def printLuas2(self) :
        print("Luas = " + str(self.hitungLuas()))

pp=PersegiPanjang()
pp.input()
pp.print()
pp.printLuas1(pp.hitungLuas())
#pp.printLuas2()

File: cpp/soal4.cpp
#include <iostream>

using namespace std;

class PersegiPanjang {
    private:
        float panjang;
        float lebar;

    public:
        PersegiPanjang() {
            this->panjang=0;
            this->lebar=0;
        }

        PersegiPanjang(float p, float l) {
            this->panjang=p;
            this->lebar=l;
        }

        void input() {
            cout << "Input Panjang: "; cin >> this->panjang;
            cout << "Input Lebar: "; cin >> this->lebar;
        }

        void setPanjang(float p) { this->panjang=p; }
        void setLebar(float l) { this->lebar=l; }

        float getPanjang() { return this->panjang; }
        float getLebar() { return this->lebar; }

        float hitungLuas() { return this->panjang * this->lebar; }
        float hitungKeliling() { return (2*this->panjang) + (2*this->lebar); }

        void print() {
            cout << "\nPanjang = " << this->panjang;
            cout << "\nLebar = " << this->lebar;
            cout << "\nLuas = " << this->hitungLuas();
            cout << "\nKeliling = " << this->hitungKeliling();
        }
};

int main() {
    PersegiPanjang *pp1 = new PersegiPanjang();
    PersegiPanjang *pp2 = new PersegiPanjang();
    PersegiPanjang *pp3 = new PersegiPanjang(15, 5);

    cout << "\nPERSEGI PANJANG 1" << endl;
    int panjang, lebar;
    cout << "Masukkan Panjang: "; cin >> panjang;
    cout << "Masukkan Lebar: "; cin >> lebar;
    pp1->setLebar(lebar);
    pp1->setPanjang(panjang);
    pp1->print();

    cout << "\nPERSEGI PANJANG 2" << endl;
    pp2->input();
    pp2->print();

    cout << "\nPERSEGI PANJANG 3" << endl;
    pp3->print();
}